benches/benchmarks.rs

```diff
@@ -57,6 +59,140 @@ fn setup_engine() -> EngineState {
 // an executable for every single one - incredibly slowly. Would be nice to figure out
 // a way to split things up again.
 
+#[divan::bench]
+fn load_standard_lib(bencher: divan::Bencher) {
+    let engine = setup_engine();
+    bencher
+        .with_inputs(|| engine.clone())
+        .bench_values(|mut engine| {
+            load_standard_library(&mut engine).unwrap();
+        })
+}
+
+#[divan::bench_group]
+mod eval_commands {
+    use super::*;
+
+    #[divan::bench(args = [100, 1_000, 10_000])]
+    fn interleave(bencher: divan::Bencher, n: i32) {
+        let mut engine = setup_engine();
+        load_standard_library(&mut engine).unwrap();
+        let commands = Spanned {
+            span: Span::unknown(),
+            item: format!("seq 1 {n} | wrap a | interleave {{ seq 1 {n} | wrap b }} | ignore")
+                .to_string(),
+        };
+
+        bencher
+            .with_inputs(|| engine.clone())
+            .bench_values(|mut engine| {
+                evaluate_commands(
+                    &commands,
+                    &mut engine,
+                    &mut nu_protocol::engine::Stack::new(),
+                    PipelineData::empty(),
+                    None,
+                )
+                .unwrap();
+            })
+    }
+
+    #[divan::bench(args = [1, 5, 10, 100, 1_000])]
+    fn for_range(bencher: divan::Bencher, n: i32) {
+        let mut engine = setup_engine();
+        load_standard_library(&mut engine).unwrap();
+        let commands = Spanned {
+            span: Span::unknown(),
+            item: format!("(for $x in (1..{}) {{ sleep 50ns }})", n).to_string(),
+        };
+        let stack = nu_protocol::engine::Stack::new();
+
+        bencher
+            .with_inputs(|| (engine.clone(), stack.clone()))
+            .bench_values(|(mut engine, mut stack)| {
+                evaluate_commands(
+                    &commands,
+                    &mut engine,
+                    &mut stack,
+                    PipelineData::empty(),
+                    None,
+                )
+                .unwrap();
+            })
+    }
+
+    #[divan::bench(args = [1, 5, 10, 100, 1_000])]
+    fn each(bencher: divan::Bencher, n: i32) {
+        let mut engine = setup_engine();
+        load_standard_library(&mut engine).unwrap();
+        let commands = Spanned {
+            span: Span::unknown(),
+            item: format!("(1..{}) | each {{|_| sleep 50ns }} | ignore", n).to_string(),
```

Comment:
Does using `ignore` here have a relevant impact?

Reply:
I am not sure, the problem is to get rid of the output that is displayed so it doesn't clutter the screen, if there is another way to do it I'm all for it.